scripts/generator/graphql_generator/csharp/SDK/iOS/iOSWebCheckout.cs.erb

```diff
@@ -36,12 +51,46 @@ namespace <%= namespace %>.SDK.iOS {
             if (message.Content == "loadedThankYou") {
                 // Do a round trip to the server to see if the purchase is solid.
                 // Invoke callbacks on cart to pass to Unity
-                OnSuccess();
+                DidCheckoutCompleteSuccessfully(CurrentCheckout.id(), (didSucceed, error) => {
+                    if (didSucceed) {
+                        OnSuccess();
+                        return;
+                    } 
+
+                    if (error != null) {
+                        OnFailure(error);
```

Comment:
I like the idea of wrapping both these in the `WebCheckoutVerificationError` since technically if we run into an HTTP error there was an error encountered while trying to verify.